### ISSUE TITLE
Accessibility of examples for interactive icons 

### ISSUE THREAD

**Issue by adildostmohamed** [FortAwesome/Font-Awesome#9566]
According to the post on how to use icons accessibly ( http://fontawesome.io/accessibility/) for icons that convey interaction and therefore should be consumed by a screen reader, the suggested example makes use of an anchor tag with a title attribute that wraps the icon that is set to aria-hidden. The use of a title attribute on an anchor element is considered to be a challenge for many user agents and should be used with caution (as per http://www.w3.org/TR/WCAG20-TECHS/H33.html). In this instance a more accessible example maybe to include a span within the anchor tag that is accessible only to a screen reader with content to provide link purpose (as is used in other examples on the font awesome site) or if this is not feasible to use the aria-label attribute rather than the title attribute as a means of proving link definition to screen reader users.

**Comment by tagliala** [comment]
Thanks for reporting this. I'm asking for feedback from folks at #6133

**Comment by tagliala** [comment]
Examples at http://fontawesome.io/examples/#accessible seem ok

```html
<a class="btn btn-default" href="path/to/settings">
  <i class="fa fa-cog" title="Settings" aria-hidden="true"></i>
  <span class="sr-only">Settings</span>
</a>
```

**Comment by adildostmohamed** [comment]
no problem, for clarity my suggestion is to update the examples used for "If an icon represents an interactive element" from:
`<a href="path/to/shopping/cart" title="View 3 items in your shopping cart">
  <i class="fa fa-shopping-cart" aria-hidden="true"></i>
</a>`

to either:

``<a href="path/to/shopping/cart">
  <i class="fa fa-shopping-cart" aria-hidden="true"></i>
 <span class="sr-only">Shopping cart</span>
</a>``

or

`<a href="path/to/shopping/cart" aria-label="Shopping cart">
  <i class="fa fa-shopping-cart" aria-hidden="true"></i>
</a>`

**Comment by adildostmohamed** [comment]
Sorry just saw your last comment, the examples used there are correct all i would suggest is perhaps updating the code snippets provided at http://fontawesome.io/accessibility/ to ensure that anyone using the code examples are consistent with your examples elsewhere

**Comment by karlgroves** [comment]
I agree with the above. The `title` attribute is less desirable than `aria-label`, especially in the case that @adildostmohamed mentioned.

**Comment by tagliala** [comment]
Is it fine to have both `title` and `aria-label` ?

> . For instance, simply adding the title attribute to the interactive element itself will be sufficient to provide an accessible alternative name for the element, **as well as providing the mouse tooltip.**

**Comment by karlgroves** [comment]
@tagliala While I understand the desire people have for the tooltip but for accessibility the title attribute is useless.  If needed, I can take a stab at documentation around this.

**Comment by adildostmohamed** [comment]
@tagliala I may be mistaken but on some screen reader / browser combinations having both a title and aria-label can result in both of those being read out to a screen reader which might lead to confusion and repeated content for a screen reader user. 

@karlgroves If users are concerned that an icon on its own may require additional context to be displayed via tooltip for additional affordance it may be desirable to create a separate tooltip element that can be triggered on hover of the anchor tag. 

In addition this same tooltip could be triggered on focus of the anchor tag which is an additional usability enhancement for users who may be using keyboard navigation and therefore wouldn't be able to see the title tooltip displayed on mouse hover, which may be a concern for users with motor disabilities. (my untested assumption here is that a title attribute only displays on hover of the element is attributed to and not on focus).

If people do decide to create an external tooltip to provide this extra information it would probably be best to hide this element from a screen reader via aria-hidden=true as they would be provided the same information via the aria-label attribute assuming the tooltip contains the same value as the aria-label. (Alternatively if this tooltip is used to create descriptive information over an above the name of the icon/action alone the aria-describedby property could be used if the tooltip contains information about the purpose of the anchor tag).

(Sorry if thats a bit incoherent, im a bit of a github beginner)

**Comment by tagliala** [comment]
@adildostmohamed what do you think about

```
<a href="path/to/shopping/cart" aria-label="View 3 items in your shopping cart">
  <i class="fa fa-shopping-cart" aria-hidden="true" title="View 3 items in your shopping cart"></i>
</a>
```

?

**Comment by karlgroves** [comment]
I like that pattern.

I also agree with @adildostmohamed that if they want a tooltip they should do that on their own.  IOW, specific to this discussion about documentation, I'd be concerned about people thinking that `title` attributes are recommended.

Similar to much of @patrickhlauke's work on Bootstrap, this is probably something that could be tackled via documenting the advantages & disadvantages of the different approaches.

**Comment by tagliala** [comment]
> Similar to much of @patrickhlauke's work on Bootstrap, this is probably something that could be tackled via documenting the advantages & disadvantages of the different approaches

Have you a reference?

Is anyone interested in submitting a pull-request with the updated doc?

**Comment by patrickhlauke** [comment]
@karlgroves is probably referring to bits and bobs i've spread across the bootstrap docs...things like the "Accesible icons" part of http://getbootstrap.com/components/#glyphicons-how-to-use, the "Always add labels" note in http://getbootstrap.com/components/#input-groups ... that sort of thing?

**Comment by tagliala** [comment]
@talbs could you take care of this change?

**Comment by dylanb** [comment]
I disagree with that WCAG statement. The reason that the title attribute is ok is that it is included in the accessible name calculation AND it is one of the most widely supported attributes for the Nuance family of products including Dragon. This means that, not only will the title attribute be supported by all ARIA compliant ATs (including all modern screen readers) but it will also work well for all recent versions of Dragon.

**Comment by talbs** [comment]
@tagliala, sure thing. Thanks for the input, folks - I'll ping you all and reference this thread on the PR.

**Comment by tagliala** [comment]
Closed via #9586
